packages/zoe/src/cleanProposal.js

```diff
@@ -128,7 +128,11 @@ const assertKeywordNotInBoth = (want, give) => {
  * must be keywords and the values must be amounts. The value of
  * `exit`, if present, must be a record of one of the following forms:
  * `{ waived: null }` `{ onDemand: null }` `{ afterDeadline: { timer
- * :Timer, deadline :bigint } }
+ * :Timer, deadline :Timestamp } }
+ * // TODO: is 'timer' meant to compare TimerBrands? If so, the type
+ * // should be TimerBrand. We don't have a Timer, we do have a
+ * // TimerService, is it meant to give the recipient a way to call
+ * // E(timer).setWakeup(deadline)?
```

Comment:
Timer in exit rule is for comparison, so a TimerBrand would be better. #5799 is about the work to pervasively refactor zoe and contracts to use TimerBrands in proposals, exit rules, and terms.

It isn't necessary that someone getting their hands on an exit rule should be able to find the current time or schedule something.

Reply:
Ok, in that case we only need the `deadline`, because it will/must be a `Timestamp` which means it will include a brand, so we don't need a separate argument. The recipient must compare the brand against the one from the timer they're using.

Incidentally, I know the API for `TimerBrand` is `brand.isMyTimerService(ts) -> bool`, but it would be dangerous for the TimerService-holding recipient to casually give away their timer authority by passing it to an arbitrary client-supplied brand. So if the recipient wants to assert something about the quality of the `deadline` timestamp, they either pass it to their TimerService APIs (which will throw if the brand doesn't match), or try to add/compare it to a Timestamp/RelativeTime they got back from their TimerService (same), or extract the brand and `===` it against their `timerService.getTimerBrand()`. I'm not sure what the best pattern would be, and maybe we will need an extra `TimeMath` helper method to assist with this sort of assertion.

Reply:
can we add that caveat to the doc for `brand.isMyTimerService(ts)`?  We try to doc similar stuff about trusting payments vs. issuers.

Reply:
added